データベース設計

tenko0126 opened the pull request:
# What
ER図の作成、およびREADMEへのデータベース設計記述

・ER図(Gyazo)：https://gyazo.com/5e87cc3ec9b869df3d20408ad35e38cb

# Why
本アプリ開発にあたるデータベース設計のため

tenko0126 committed:
データベース設計

yuri240 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+| Column             | Type   | Options     |
+| ------------------ | ------ | ----------- |
+| nickname           | string | null: false |
+| email              | string | null: false |
```
emailカラムにユニーク制約を適応させましょう。

■理由
要件においてメールアドレスは一意性であることが求められているためです。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+| sales-status        | string     | null: false                    |
+| shipping-fee-status | string     | null: false                    |
+| prefecture          | string     | null: false                    |
+| scheduled-delivery  | date       | null: false                    |
```
**商品の状態, 配送料の負担, 発送元の地域,発送までの日数,カテゴリーのカラム**

上記カラムはactivehashを用いた実装になるため、activehashのルールに沿ったカラム名と型にしましょう。
activehashにつきましては以下カリキュラムをご参考にいただければと思います。
```
Lesson1 最終課題補足カリキュラム
カテゴリーの選択を工夫しよう
```

yuri240 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+| info                | text       | null: false                    |
+| category            | string     | null: false                    |
+| sales-status        | string     | null: false                    |
+| shipping-fee-status | string     | null: false                    |
```
命名規則上、カラム名に`-`は使用できないため、アンダースコア`_`を使用しましょう！

yuri240 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+| Column       | Type       | Options                        |
+| ------------ | ---------- | ------------------------------ |
+| postal-code  | string     | null: false                    |
+| prefecture   | string     | null: false                    |
```
```
| prefecture   | string     | null: false                    |
```
こちらの都道府県カラムは、
active_hashのデータとして使い回すことができますので、
itemsテーブルの都道府県カラムと同じものを使用しましょう。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+| prefecture   | string     | null: false                    |
+| city         | string     | null: false                    |
+| addresses    | string     | null: false                    |
+| building     | string     | null: false                    |
```
```
| building     | string     | null: false                    |
```

建物名のカラムは、**任意項目**なので
`null:false`を外してあげましょう！

よって、
`building`の`Options`は**空欄**にしていただければ大丈夫です。

yuri240 commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+| postal-code  | string     | null: false                    |
+| prefecture   | string     | null: false                    |
+| city         | string     | null: false                    |
+| addresses    | string     | null: false                    |
```
`| addresses    | string     | null: false                    |`
カラム名は**単数形**の単語を使用しましょう！

yuri240 reviewed (commented):
レビュー依頼ありがとうございます！
コメントさせていただきましたので、ご確認お願いいたします。

また、次回ご提出の際にはER図も修正が必要となります。

お手数ではありますが、
修正後のER図を再度こちらのプルリクエストに添付していただくようお願いいたします。

tenko0126 committed:
データベース設計_Δ1

・カラム名の「-(ハイフン)」を「_(アンダーバー)」に修正
・カラム名が複数形であるものを単数形に修正
・active_hashの使用を前提にカラム名・型を修正
・active_hash使用のためGemfile追記
上記の内容に応じてER図修正

tenko0126 commented:
コードレビューおよび修正点のご指摘ありがとうございました。
修正しましたので、改めてレビューお願いします。

■ER図修正版
https://gyazo.com/d69b05272d21f30d820922fdd8aa1cf4

TakashiKozuma commented on README.md:
```diff
@@ -22,3 +22,70 @@ Things you may want to cover:
+| sales_status_id        | integer    | null: false                    |
+| shipping_fee_status_id | integer    | null: false                    |
+| prefecture_id          | integer    | null: false                    |
+| scheduled_delivery     | date       | null: false                    |
```
```rb
| scheduled_delivery | date | null: false |
```
こちらもアクティブハッシュを用いたカラムなので、カラム名は末尾に_idを付け、stringではなくintegerで実装しましょう。
後の実装で、アソシエーションを用いてデータを取り出すときに必要になってきます。

TakashiKozuma reviewed (commented):
レビュー依頼ありがとうございます！
修正点確認致しました。
コメントを追記いたしました。
ご確認お願いいたします。

修正後はコミット・プッシュのし忘れがないかご確認の上、再度フォームからレビュー依頼をお願い致します。

tenko0126 committed:
データベース設計Δ2

・READMEを修正
・ER図(ER.dio)を修正

tenko0126 commented:
再度のコードレビュー/修正指摘をありがとうございました。
内容修正しましたので、改めてレビューお願いいたします。

■ER図修正版
https://gyazo.com/f188c576d8caec879fa1c900512517e3

yuri240 reviewed (commented):
修正とレビュー依頼ありがとうございます。

しっかりと設計できていることが確認できましたので、
**LGTM**とさせていただきます！

こちらのブランチは忘れずにMergeを行っていただき、
ローカルのMasterにPullしておきましょう。

データベース設計の実装、お疲れ様でした。
引き続き、応援しております。
